src/main/java/com/sequenceiq/cloudbreak/service/stack/connector/aws/AwsConnector.java

```diff
@@ -157,76 +165,91 @@ public void rollback(Stack stack, Set<Resource> resourceSet) {
-        AmazonCloudFormationClient client = awsStackUtil.createCloudFormationClient(awsTemplate.getRegion(), awsCredential);
+        AmazonCloudFormationClient client = awsStackUtil.createCloudFormationClient(Regions.valueOf(stack.getRegion()), awsCredential);
         String stackName = String.format("%s-%s", stack.getName(), stack.getId());
-        boolean spotPriced = awsTemplate.getSpotPrice() == null ? false : true;
         List<Parameter> parameters = new ArrayList<>(Arrays.asList(
-                new Parameter().withParameterKey("SSHLocation").withParameterValue(awsTemplate.getSshLocation()),
+                new Parameter().withParameterKey("SSHLocation").withParameterValue("0.0.0.0/0"),
                 new Parameter().withParameterKey("CBUserData").withParameterValue(userData),
                 new Parameter().withParameterKey("StackName").withParameterValue(stackName),
                 new Parameter().withParameterKey("StackOwner").withParameterValue(awsCredential.getRoleArn()),
-                new Parameter().withParameterKey("InstanceCount").withParameterValue(stack.getNodeCount().toString()),
-                new Parameter().withParameterKey("InstanceType").withParameterValue(awsTemplate.getInstanceType().toString()),
                 new Parameter().withParameterKey("KeyName").withParameterValue(awsCredential.getKeyPairName()),
-                new Parameter().withParameterKey("AMI").withParameterValue(awsTemplate.getAmiId()),
-                new Parameter().withParameterKey("VolumeSize").withParameterValue(awsTemplate.getVolumeSize().toString()),
-                new Parameter().withParameterKey("VolumeType").withParameterValue(awsTemplate.getVolumeType().toString())));
-        if (spotPriced) {
-            parameters.add(new Parameter().withParameterKey("SpotPrice").withParameterValue(awsTemplate.getSpotPrice().toString()));
-        }
+                new Parameter().withParameterKey("AMI").withParameterValue(prepareAmis().get(Regions.valueOf(stack.getRegion()).getName()))
+        ));
         CreateStackRequest createStackRequest = createStackRequest()
                 .withStackName(stackName)
-                .withTemplateBody(cfTemplateBuilder.build("templates/aws-cf-stack.ftl", awsTemplate.getVolumeCount(), spotPriced))
+                .withTemplateBody(cfTemplateBuilder.build("templates/aws-cf-stack.ftl",
+                        spotPriceNeeded(stack.getInstanceGroups()), stack.getTemplateSetAsList()))
                 .withNotificationARNs((String) setupProperties.get(SnsTopicManager.NOTIFICATION_TOPIC_ARN_KEY))
                 .withParameters(parameters);
         client.createStack(createStackRequest);
+
+
         Set<Resource> resources = new HashSet<>();
-        resources.add(new Resource(ResourceType.CLOUDFORMATION_STACK, stackName, stack));
+        resources.add(new Resource(ResourceType.CLOUDFORMATION_STACK, stackName, stack, Lists.newArrayList(stack.getInstanceGroups()).get(0).getGroupName()));
```

Comment:
cannot be the instanceGroup simply null if it has no meaning for a CF stack resource?